eng/pipelines/cpp-nightly.yml

```diff
@@ -0,0 +1,74 @@
+
+jobs:
+  - job:
+    displayName: Nightly vcpkg PR
+    pool:
+      name: azsdk-pool-mms-win-2019-general
+      vmImage: MMS2019
+
+    variables:
+      # The branch in the azure-sdk/vcpkg repo that is updated nightly with the
+      # latest set of ports
+      TargetPrBranch: azure-sdk-for-cpp-nightly-verification
+
+    steps:
+      - checkout: self
+
+      - template: /eng/pipelines/templates/steps/generate-nightly-branch-name.yml
+
+      - template: /eng/pipelines/templates/steps/vcpkg-clone.yml
+        parameters:
+          Workspace: $(Pipeline.Workspace)
+          RepoOwner: azure-sdk
+          SkipCheckout: true
+
+      - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
+        parameters:
+          WorkingDirectory: $(Pipeline.Workspace)/vcpkg
+
+      # We have to set git identity because git merge will create a merge commit
+      # in some cases and event when specifying --no-commit.
+      - pwsh: |
+          Write-Host 'git config user.name "azure-sdk"'
+          git config user.name "azure-sdk"
+          Write-Host 'git config user.email "azuresdk@microsoft.com"'
+          git config user.email "azuresdk@microsoft.com"
+
+          Write-Host "git checkout $(NightlyBranchName)"
+          git checkout $(NightlyBranchName)
```

Comment:
I think all you need to do is checkout the nightly branch and then mark HasChanges=true